src/main/java/org/opensearch/security/dlic/rest/api/RolesMappingApiAction.java

```diff
@@ -150,8 +153,35 @@ protected Endpoint getEndpoint() {
     }
 
     @Override
-    protected AbstractConfigurationValidator getValidator(RestRequest request, BytesReference ref, Object... param) {
-        return new RolesMappingValidator(request, isSuperAdmin(), ref, this.settings, param);
+    protected RequestContentValidator createValidator(final Object... params) {
+        return RequestContentValidator.of(new RequestContentValidator.ValidationContext() {
+            @Override
+            public Object[] params() {
+                return params;
+            }
+
+            @Override
+            public Settings settings() {
+                return settings;
+            }
+
+            @Override
+            public Set<String> mandatoryOrKeys() {
```

Comment:
Nice! I like this vs writing this as an if statement checking if at least one is nonNull.